src/test/java/net/imagej/legacy/BatchModeTest.java

```diff
@@ -0,0 +1,104 @@
+package net.imagej.legacy;
+
+import ij.IJ;
+import ij.WindowManager;
+import net.imagej.Dataset;
+import net.imagej.DatasetService;
+import net.imagej.axis.Axes;
+import net.imagej.axis.AxisType;
+import net.imagej.patcher.LegacyInjector;
+import net.imglib2.type.numeric.integer.UnsignedByteType;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.scijava.Context;
+import org.scijava.ItemIO;
+import org.scijava.command.Command;
+import org.scijava.plugin.Parameter;
+import org.scijava.plugin.Plugin;
+import org.scijava.ui.UIService;
+
+import java.awt.*;
+
+import static org.junit.Assert.assertArrayEquals;
+import static org.junit.Assume.assumeFalse;
+
+/**
+ * Test if a {@link Command} that returns a {@link Dataset} behaves as expected
+ * when a IJ1 macro is run in batch mode.
+ */
+public class BatchModeTest {
+
+	static {
+		LegacyInjector.preinit();
+	}
+
+	private Context context;
+
+	@Before
+	public void setUp() {
+		// only run the tests if a graphics environment is present
+		assumeFalse(GraphicsEnvironment.isHeadless());
```

Comment:
Shouldn't running a macro work *the same* whether or not in headless mode? If it fails in headless, it's a bug (either here or in `ij1-patcher`), no?

Reply:
A macro should run the same way, if headless in batch mode or with UI. But that's currently not the case. This PR fixes only a small part of the problem. A macro running in batch mode will now have access to the output images of a scijava command. But this works only if the legacy UI is visible.